examples/daisyworld.jl

```diff
@@ -48,249 +60,269 @@ mutable struct Daisy <: AbstractAgent
+
+# Notice that we want to ensure that the `Land` patches are always plotted first.
+plotsched = by_type((Land, Daisy), false)
+
+plotkwargs = (
+    ac = daisycolor, am = daisyshape, as = daisysize,
+    scheduler = plotsched, aspect_ratio = 1, size = (600, 600), showaxis = false,
 )
-plot(p1, p2, p3, layout = (3, 1), size = (500, 800))
+
+p = plotabm(model; plotkwargs...)
+
+# And after a couple of steps
+step!(model, agent_step!, model_step!, 5)
+p = plotabm(model; plotkwargs...)
+
+# Let's do some animation now
+Random.seed!(165) # hide
+model = daisyworld()
+anim = @animate for i in 0:30
+    p = plotabm(model; plotkwargs...)
+    title!(p, "step $(i)")
+    step!(model, agent_step!, model_step!)
+end
+gif(anim, "daisyworld.gif", fps = 3)
+
+# Running this animation for longer hints that this world achieves quasi-equilibrium
+# for some input parameters, where one `breed` does not totally dominate the other.
+# Of course we can check this easily through data collection.
+# Notice that here we have to define a function `breed` that gives the daisy's `breed`
+# field. We cannot use just `:breed` to automatically find it, because in this mixed
+# agent model, the `Land` doesn't have any `breed`.
+# %% # src
+black(y) = count(x -> x == :black, y)
+white(y) = count(x -> x == :white, y)
+breed(a) = a isa Daisy ? a.breed : :land
+adata = [(breed, black), (breed, white)]
+
+Random.seed!(165) # hide
+model = daisyworld(; solar_luminosity = 1.0)
+
+agent_df, model_df = run!(model, agent_step!, model_step!, 1000; adata = adata)
+
+p = plot(agent_df[!, :step], agent_df[!, :black_breed], label = "black")
+plot!(p, agent_df[!, :step], agent_df[!, :white_breed], label = "white")
+plot!(p; xlabel = "tick", ylabel = "daisy count")
+
+# ## Time dependent dynamics
+# %% #src
+
+# To use the time-dependent dynamics we simple use the keyword `scenario = :ramp` during
+# model creation. However, we also want to see how the planet surface temperature changes
+# and would be nice to plot solar luminosity as well.
+# Thus, we define in addition
+gettemperature(a) = a isa Land ? a.temperature : missing
+meantemperature(x) = mean(skipmissing(x))
+adata = [(breed, black), (breed, white), (gettemperature, meantemperature)]
+
+# And, to have it as reference, we also record the solar luminosity value
+mdata = [:solar_luminosity]
+
+# And we run (and plot) everything
+Random.seed!(165) # hide
+model = daisyworld(solar_luminosity = 1.0, scenario = :ramp)
+agent_df, model_df = run!(model, agent_step!, model_step!, 1000; adata = adata, mdata = mdata)
+
+p = plot(agent_df[!, :step], agent_df[!, :black_breed], label = "black")
+plot!(p, agent_df[!, :step], agent_df[!, :white_breed], label = "white")
+plot!(p; xlabel = "tick", ylabel = "daisy count")
+
+p2 = plot(agent_df[!, :step], agent_df[!, aggname(adata[3])], ylabel = "temperature")
+p3 = plot(model_df[!, :step], model_df[!, :solar_luminosity], ylabel = "L", xlabel = "ticks")
+
+plot(p, p2, p3, layout = (3, 1))
```

Comment:
Probably worth adding `, size = (500, 800)` to the arguments here. Otherwise the graph is a little small.